internal/tracker/tracker.go

```diff
@@ -23,22 +23,47 @@ import (
 	"time"
 
 	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
 	"github.com/stuart-warren/yamlfmt"
+	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/hacbs-contract/ec-cli/internal/image"
 )
 
-type Record struct {
+const (
+	pipelineCollection = "pipeline-bundles"
+	taskCollection     = "task-bundles"
+)
+
+type bundleRecord struct {
 	Digest      string    `json:"digest"`
 	EffectiveOn time.Time `json:"effective_on"`
 	Tag         string    `json:"tag"`
 	Repository  string    `json:"-"`
 	Collection  string    `json:"-"`
 }
 
-type Tracker map[string]map[string][]Record
+type commonTasksRecord struct {
+	Tasks       []string  `json:"tasks"`
+	EffectiveOn time.Time `json:"effective_on"`
+	initialized bool
+}
+
+func (r *commonTasksRecord) updateTasksIntersection(newTasks sets.String) {
+	if !r.initialized {
```

Comment:
Perhaps, remove the need for `initialized
```suggestion
	if len(r.Tasks) == 0 {
```

Reply:
Actually... a pipeline that has no tasks is probably something we want to ignore. I'll switch the logic (and get rid of `initialized`).

Reply:
Done.